dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -914,37 +961,31 @@ default String getDockerToken(String registryDockerPath, String repo) throws URI
      * @param repo
      * @param digest SHA256 digest of the blob to download
      * @return HttpResponse
-     * @throws URISyntaxException
-     * @throws IOException
-     * @throws InterruptedException
      */
-    default HttpResponse<String> getDockerBlob(String token, String registryDockerPath, String repo, String digest)
-            throws URISyntaxException, IOException, InterruptedException {
+    default Optional<HttpResponse<String>> getDockerBlob(String token, String registryDockerPath, String repo, String digest) {
         // Ex: https://ghcr.io/v2/<repo>/blobs/<digest>
         String getBlobURL = String.format("https://%s/v2/%s/blobs/%s", registryDockerPath, repo, digest);
-        HttpRequest request = HttpRequest.newBuilder()
-                .uri(new URI(getBlobURL))
-                .header(HttpHeaders.AUTHORIZATION, String.join(" ", JWT_SECURITY_DEFINITION_NAME, token))
-                .GET()
-                .build();
-
-        // This endpoint may issue a 307 redirect to another service to download the blob
-        return HttpClient.newBuilder()
-                .followRedirects(HttpClient.Redirect.NORMAL) // NORMAL: Always redirect, except from HTTPS URLs to HTTP URLs
-                .proxy(ProxySelector.getDefault())
-                .build()
-                .send(request, HttpResponse.BodyHandlers.ofString());
-    }
-
-    default Set<Image> getImagesFromGitHubContainerRegistry(final String repo, final DockerSpecifier specifierType, final String specifierName) {
-        Set<Image> gitHubContainerRegistryImages = new HashSet<>();
-        Optional<Set<Image>> images = getImages(Registry.GITHUB_CONTAINER_REGISTRY, repo, specifierType, specifierName);
-
-        if (images.isPresent()) {
-            gitHubContainerRegistryImages.addAll(images.get());
+        HttpResponse<String> blobResponse;
+        try {
+            HttpRequest request = HttpRequest.newBuilder().uri(new URI(getBlobURL)).header(HttpHeaders.AUTHORIZATION, String.join(" ", JWT_SECURITY_DEFINITION_NAME, token)).GET().build();
+            // This endpoint may issue a 307 redirect to another service to download the blob
+            blobResponse = HttpClient.newBuilder()
+                    .followRedirects(HttpClient.Redirect.NORMAL) // NORMAL: Always redirect, except from HTTPS URLs to HTTP URLs
+                    .proxy(ProxySelector.getDefault())
+                    .build()
+                    .send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (URISyntaxException | IOException | InterruptedException ex) {
+            LOG.error("Could not send blob request GET {}", getBlobURL, ex);
+            return Optional.empty();
         }
 
-        return gitHubContainerRegistryImages;
+        if (blobResponse.statusCode() != HttpStatus.SC_OK) {
+            Map<String, List<Map<String, String>>> errorMap = GSON.fromJson(blobResponse.body(), Map.class);
```

Comment:
Same as other comments. Also just noticed that all 3 places where you do the error handling are near duplicates (I think), so you could create a common method.